docs/reference/technology/connectivity/cellular.md

```diff
@@ -0,0 +1,159 @@
+
+    set_ciot_optimization_config(Supported_UE_Opt supported_opt,
+                                 Preferred_UE_Opt preferred_opt)
+
+### Optimize for power consumption
+
+The `CellularPower` class has methods to optimize power saving. The `set_powerl_level()` offers flexibility to control the reception and transmission power levels. In addition, 3GPP has specified advanced power optimizations that are useful for celluar IoT devices: Power Save Mode (PSM) and extended Discontinuous Reception (eDRX).
+
+#### PSM - Power Save Mode
+
+    opt_power_save_mode(int periodic_time, int active_time)
+
+The new 4G modems implementing specifications release 13 and later include PSM. PSM allows the application to tell the modem and network that it does not expect any data in a given time interval, the `periodic_time`. The modem and network can optimize the sleep state and network resource reservations based on this information. During the PSM time, nothing can contact the device from the network side. The application can still send at any time. PSM time can be hours, days or weeks.
+
+The `active_time` tells how many seconds the device waits to receive messages after it has sent data, or when the periodic time has lapsed. If the device sends a report to the management system, the device waits in listen state for the configured amount of time. Listening keeps the radio on in the modem, so it consumes more power than idle state.
+
+This feature offers great power savings for periodically reporting devices. Between reports, the device is in PSM state with the modem in the deep sleep state. The latency of contacting the device is the PSM period.
+
+PSM configuration is negotiated with the network, and the actual PSM time that network has accepted may differ from that requested.
+
+#### eDRX - extended Discontinuous Reception
+
+    opt_receive_period(int mode, EDRXAccessTechnology act_type, uint8_t edrx_value)
+
+eDRX tells how long the device sleeps in continuous connection. The device needs to be able to receive data, but it can tell the network that it checks for incoming messages, for example only every 200 seconds. It can receive messages but only at the given times to allow battery saving sleep periods. These time values are greatly extended compared to normal 4G data transmission – hence the name.
+
+This feature serves devices that need smaller latencies. A connection is kept open all the time, but the modem achieves a sleep state between the reception times. The data connection, if using IPv4, may need periodic keep alive messages to keep the network address translation mapping valid.
+
+eDRX configuration is given to the modem. It is negotiated with the network, and the time accepted by the network may differ from the requested time. Availability of this optimization depends on the cellular network.
```

Comment:
Query: Who or what gives eDRX configuration to the modem? Also, who or what negotiates it with the network?

Reply:
Clarified that An application gives eDRX configuration to the modem in https://github.com/AriParkkila/Handbook/commit/d9e98f640a283cdbdc9c44c2daba43c6cbcf01d6